src/api/channels.js

```diff
@@ -207,6 +224,22 @@ export async function updateChannel (ctx, channelId) {
   // Set the user updating the channel for auditing purposes
   channelData.updatedBy = utils.selectAuditFields(ctx.authenticated)
 
+  if (!utils.isNullOrWhitespace(channelData.type) && utils.isNullOrEmpty(channelData.methods)) {
+    // Empty the methods if the type has changed from http
+    if (channelData.type !== 'http') {
+      channelData.methods = []
```

Comment:
Why are we only interested in checking the methods for http channels? Why not on the others?